### PR TITLE
DALTON: fix parsing electronic transition coefficients for non-DFT wavefunctions

### PR DESCRIPTION
**Pull request by berquist**
Closes https://github.com/cclib/cclib/issues/677. With this fix, more tests can be enabled.

This PR should go in **before** https://github.com/cclib/cclib-data/pull/106.

### CODE REVIEW

File: cclib/parser/daltonparser.py [hunk 1]
```diff
@@ -1148,8 +1151,9 @@ def extract(self, inputfile, line):
```
Comment (langner) [inline cclib/parser/daltonparser.py]: Consider adding a comment with an example above this block.

File: test/data/testTD.py [hunk 2]
```diff
@@ -24,7 +24,6 @@ class GenericTDTest(unittest.TestCase):
     number = 5
     expected_l_max = 41000
 
-    @skipForParser('DALTON', 'etoscs are not parsed')
```
Comment (langner) [inline test/data/testTD.py]: Was this failing before?

File: test/regression.py [hunk 3]
```diff
@@ -165,6 +165,16 @@ def testADF_ADF2016_fa2_adf_out(logfile):
 
 # DALTON #
 
+def testDALTON_DALTON_2013_dvb_td_normalprint_out(logfile):
+    """This original unit test prints a DFT-specific version of the excitation
```
Comment (langner) [inline test/regression.py]: Could you reiterate why we do not parse the DFT-specific version?

Comment (berquist) [reply test/regression.py]: The field that was parsed (https://github.com/cclib/cclib/pull/680/files#diff-88f10a78215b8cf004128ef198e5762bL1197) is incorrect, and the correct field (the scaled one) that matches what other programs print isn't available in the DFT-specific output.

Comment (langner) [reply test/regression.py]: Could you make this clear in the docstring here, and/or in the parser code? I think these types of things are worth documenting, so we don't trip over them again in the future.

**Comment by langner** [comment]
Redirecting this to the `release-1.6.1` branch as well.

File: test/regression.py (hunk 3 above)
Comment (berquist) [reply test/regression.py]: Done in https://github.com/cclib/cclib/pull/680/commits/bfd9a8d8bbaa1b34e2af4d57e30174fd7e34d964

Comment (langner) [reply test/regression.py]: Thanks!

**Pull request merged**